src/statement.rs

```diff
@@ -57,21 +57,28 @@ impl<'l> Statement<'l> {
         unsafe { ffi::sqlite3_column_count(self.raw.0) as usize }
     }
 
-    fn column_name(&self, idx: c_int) -> &str {
-        let cname = unsafe { 
-            CStr::from_ptr(ffi::sqlite3_column_name(self.raw.0, idx))
-        };
-        str::from_utf8(cname.to_bytes()).unwrap()
+    /// Return the column name from the statement
+    ///
+    /// The leftmost column has the index 0.
+    fn column_name(&self, i: usize) -> Result<String> {
```

Comment:
I’m thinking that, if these two methods are introduced, `columns` will look as an outlier. It should probably be renamed to `column_count`.